astropy/io/ascii/tdat.py

```diff
@@ -0,0 +1,815 @@
+            if len(keywords["table_name"]) > 20:
+                warn(
+                    "'table_name' is too long, truncating to 20 characters",
+                    TdatFormatWarning,
+                )
+            lines.append(f"table_name = {keywords.pop('table_name')[:20]}")
+        else:
+            warn(
+                "'table_name' must be specified\n"  # noqa: ISC003
+                + f"{_STD_MSG}\n"
+                + "This should be specified in the Table.meta.\n"
+                + "default value of 'astropy_table' being assigned.",
+                TdatFormatWarning,
+            )
+            lines.append("table_name = astropy_table")
+
+        # loop through optional table keywords
+        table_keywords = [
+            "table_description",
+            "table_document_url",
+            "table_security",
+        ]
+        table_keywords = [kw for kw in table_keywords if kw in keywords.keys()]
+        for kw in ["record_delimiter", "field_delimiter"]:
+            val = keywords.pop(kw, None)
+            if val is not None:
+                warn(TdatFormatWarning(f"Skipping deprecated keyword: {kw}."))
+
+        for key in table_keywords:
+            if key == "table_description":
+                if len(keywords[key]) > 80:
+                    warn(
+                        "'table_description' is too long, truncating to 80 characters",
+                        TdatFormatWarning,
+                    )
+                new_desc = keywords.pop(key)
+                new_desc = new_desc[:80]
+                lines.append(f"{key} = {new_desc}")
+            elif key in keywords:
+                lines.append(f"{key} = {keywords.pop(key)}")
+
+        # add table columns as fields
+        lines.append("#")
+        lines.append("# Table Parameters")
+        lines.append("#")
+        for col in self.cols:
+            if str(col.info.dtype) in self._dtype_dict_out:
+                ctype = self._dtype_dict_out[str(col.info.dtype)]
+            elif "int" in str(col.info.dtype):
+                ctype = "int4"
+            elif "float" in str(col.info.dtype):
+                ctype = "float8"
+            elif "<U" in str(col.info.dtype):
+                ctype = f"char{str(col.info.dtype).rsplit('<U', maxsplit=1)[-1]}"
+            else:
+                raise TdatFormatError(
+                    f'Unrecognized data type `{col.info.dtype}` for column "{col.info.name}".'
+                )
+                # ctype = f"{col.info.dtype}"
+            field_line = f"field[{col.info.name}] = {ctype}"
+
+            col_info_meta = col.info.meta or {}
+            if col.info.format is not None:
+                field_line += f":{col.info.format}"
+            if col.info.unit is not None:
+                if " " in f"{col.info.unit}":
+                    unit_str = f"{col.info.unit:cds}"
+                else:
+                    unit_str = f"{col.info.unit}"
+                field_line += f"_{unit_str}"
```

Comment:
For consistency and simplicity, always output using CDS unit formatting. Then the `Tdat` class docstring should mention that units are always written in CDS format.
```suggestion
                field_line += f"_{col.info.unit:cds}"
```